CMakeLists.txt

```diff
@@ -185,16 +185,18 @@ LINK_DIRECTORIES(${Boost_LIBRARY_DIRS})
 # up the targets (Exported) that it needs. We just need to find where
 # HDF5 is installed.
 
-if("${HDF5_INSTALL}" STREQUAL "")
-    set(HDF5_INSTALL  $ENV{HDF5_INSTALL})
-endif()
+find_package(HDF5)
```

Comment:
We put the HDF5_INSTALL environment variable check first because most of the time we need to over ride which HDF5 is used to compile. Once the HDF5_INSTALL variable is used, the find_package() for our custom FindHDF5.cmake file will use that variable to find the correct HDF5 version.